index.js

```diff
@@ -1,54 +1,65 @@
 /*! load-ip-set. MIT License. WebTorrent LLC <https://webtorrent.io/opensource> */
 import fs from 'fs'
-import get from 'simple-get'
+import fetch from 'cross-fetch-ponyfill'
 import IPSet from 'ip-set'
 import { Netmask } from 'netmask'
 import once from 'once'
 import split from 'split'
 import zlib from 'zlib'
+import queueMicrotask from 'queue-microtask'
 
 // Match single IPs and IP ranges (IPv4 and IPv6), with or without a description
 const ipSetRegex = /^\s*(?:[^#].*?\s*:\s*)?([a-f0-9.:]+)(?:\s*-\s*([a-f0-9.:]+))?\s*$/
 
 // Match CIDR IPv4 ranges in the form A.B.C.D/E, with or without a description
 const cidrRegex = /^\s*(?:[^#].*?\s*:\s*)?([0-9.:]+)\/([0-9]{1,2})\s*$/
 
-function loadIPSet (input, opts, cb) {
+async function loadIPSet (input, opts, cb) {
   if (typeof opts === 'function') return loadIPSet(input, {}, opts)
   cb = once(cb)
 
   if (Array.isArray(input) || !input) {
-    process.nextTick(() => {
+    queueMicrotask(() => {
       cb(null, new IPSet(input))
     })
   } else if (/^https?:\/\//.test(input)) {
-    opts.url = input
-    get(opts, (err, res) => {
-      if (err) return cb(err)
-      onStream(res)
-    })
+    let res = null
+    try {
+      res = await fetch(input, opts)
```

Comment:
Is there a reason we're no longer processing this as a stream? 

Loading large lists directly into memory may cause memory problems.

Reply:
I didn't see much reason to, the list would need to be INSANELY big for it to cause memory issues, like a 1_000_000_000 characters before it actually becomes an issue, so it's not too big of a concern tbh?

Reply:
Sounds like a reasonable assumption as long as lists are correctly formatted in CDIR notation.